### PR TITLE
[ISSUE-22] Fix private variable naming scheme

### PR DESCRIPTION
**Pull request by aiola**

### CODE REVIEW

File: cgnal/core/data/layer/__init__.py [hunk 1]
```diff
@@ -113,10 +113,11 @@ def retrieveGenerator(self, condition: Any, sort_by: Any) -> IterGenerator[T]:
         :param Any sort_by: sorting options
         :return IterGenerator[T]: iterator generator
         """
-        def __iterator__():
+
+        def __iterator():
```
Comment (NicolaDonelli) [inline cgnal/core/data/layer/__init__.py:117]: Why using two leading underscores here while using a single underscore in most other cases?

File: cgnal/core/data/layer/mongo/archivers.py [hunk 2]
```diff
@@ -73,9 +73,9 @@ def archiveOne(self, obj: T) -> UpdateResult:
         :param obj: document to archive
         :return: an instance of :class:`pymongo.results.UpdateResult` with update operation's results
         """
-        return self.__insert__(obj)
+        return self.__insert(obj)
 
-    def __insert__(self, obj: T) -> UpdateResult:
+    def __insert(self, obj: T) -> UpdateResult:
```
Comment (NicolaDonelli) [inline cgnal/core/data/layer/mongo/archivers.py:78]: Same as above

File: cgnal/core/data/model/ml.py [hunk 3]
```diff
@@ -113,7 +113,7 @@ class MultiFeatureSample(Sample[List[np.ndarray], LabType]):
     """Class representing an observation defined by a nested list of arrays."""
 
     @staticmethod
-    def __check_features__(features: List[np.ndarray]) -> None:
+    def __check_features(features: List[np.ndarray]) -> None:
```
Comment (NicolaDonelli) [inline cgnal/core/data/model/ml.py:116]: Same as above

File: cgnal/core/utils/decorators.py [hunk 4]
```diff
@@ -49,13 +49,13 @@ class Cached(object):
     """Class to cache results and export them as pickle to be later reloaded."""
 
     @property
-    def __cache__(self):
+    def __cache(self):
```
Comment (NicolaDonelli) [inline cgnal/core/utils/decorators.py:52]: Same here

**Review by NicolaDonelli — CHANGES_REQUESTED** [review]
Everything seems functionally fine, but why did you choose to use two different nomenclatures (one with a double leading underscore, the other one with a single leading underscore)?
I would prefer to keep the same nomenclature everywhere (no matter which one...)

**Comment by NicolaDonelli** [comment]
Ok, got it. 
But, since in python nothing is truly private, we never put too much attention name mangling. 
Having said that, I am not against adopting this differentiation but since, in principle, we never wanted to do name mangling (except for magic methods, we always wanted to simply point out that a given method should not be directly accessed by the user), I would be cautious in differentiating between these two types of hidden methods.

**Comment by aiola** [comment]
> Ok, got it. But, since in python nothing is truly private, we never put too much attention name mangling. Having said that, I am not against adopting this differentiation but since, in principle, we never wanted to do name mangling (except for magic methods, we always wanted to simply point out that a given method should not be directly accessed by the user), I would be cautious in differentiating between these two types of hidden methods.

Ok, I fixed it. All private methods have now a single leading underscore.

**Review by NicolaDonelli — APPROVED** [review]
Fine by me

**Pull request merged**